variable/test/variable_special_values_test.cpp

```diff
@@ -0,0 +1,211 @@
+#include <gtest/gtest.h>
+#include <vector>
+
+#include "fix_typed_test_suite_warnings.h"
+#include "test_macros.h"
+
+#include "scipp/core/element/special_values.h"
+#include "scipp/variable/arithmetic.h"
+#include "scipp/variable/special_values.h"
+#include "scipp/variable/variable.h"
+
+using namespace scipp;
+using namespace scipp::core;
+using namespace scipp::variable;
+
+template <typename T> constexpr auto values_for_special_value_tests() {
+  return std::vector{static_cast<T>(0.0),
+                     static_cast<T>(-1.23),
+                     static_cast<T>(3e4),
+                     std::numeric_limits<T>::quiet_NaN(),
+                     std::numeric_limits<T>::signaling_NaN(),
+                     std::numeric_limits<T>::infinity(),
+                     -std::numeric_limits<T>::infinity()};
+}
+
+template <typename T>
+class VariableSpecialValueTest : public ::testing::Test {};
+using FloatTypes = ::testing::Types<double, float>;
+TYPED_TEST_SUITE(VariableSpecialValueTest, FloatTypes);
+
+TYPED_TEST(VariableSpecialValueTest, isnan) {
+  for (TypeParam x : values_for_special_value_tests<TypeParam>()) {
+    for (auto u : {units::dimensionless, units::m}) {
+      const auto res = isnan(x * u);
+      EXPECT_EQ(res.template value<bool>(), element::isnan(x));
+      EXPECT_EQ(res.unit(), units::dimensionless);
+    }
+  }
+}
+
+TYPED_TEST(VariableSpecialValueTest, isinf) {
+  for (TypeParam x : values_for_special_value_tests<TypeParam>()) {
+    for (auto u : {units::dimensionless, units::m}) {
+      const auto res = isinf(x * u);
+      EXPECT_EQ(res.template value<bool>(), element::isinf(x));
+      EXPECT_EQ(res.unit(), units::dimensionless);
+    }
+  }
+}
+
+TYPED_TEST(VariableSpecialValueTest, isfinite) {
+  for (TypeParam x : values_for_special_value_tests<TypeParam>()) {
+    for (auto u : {units::dimensionless, units::m}) {
+      const auto res = isfinite(x * u);
+      EXPECT_EQ(res.template value<bool>(), element::isfinite(x));
```

Comment:
Could the Windows issue be a problem with the space here in a macro?

Reply:
Do macros process spaces? I thought they only care about commas.

Reply:
I don't know, since I didn't write *this* macro.

Reply:
The latest commits removed this code. So this particular error is not a problem anymore. However, there is a new one on Windows, see below.